Jobs.Fetcher.Twitter/Jobs/Ads/CampaignsQuery.cs

```diff
@@ -37,9 +37,14 @@ public class CampaignsQuery : AbstractTwitterFetcher {
             var adsAccountIds = DbReader.GetAdsAccountIds(username, dbContext);
 
             void ProccessCampaignResult(string accountId, ITwitterRequestIterator<CampaignsResponse, string> iterator) {
-                while (!iterator.Completed) {
-                    var campaignsPage = iterator.NextPageAsync().GetAwaiter().GetResult();
-                    DbWriter.WriteCampaigns(accountId, campaignsPage.Content, dbContext, GetLogger());
+                try {
+                    while (!iterator.Completed) {
+                        var campaignsPage = iterator.NextPageAsync().GetAwaiter().GetResult();
+                        DbWriter.WriteCampaigns(accountId, campaignsPage.Content, dbContext, GetLogger());
+                    }
+                }catch (Exception e) {
```

Comment:
A suggestion for future improvement: separating this between errors that could arise from external library `FlycatcherAds` and errors which are "our fault", e.g., database integrity check fails

Reply:
FlycatcherAds has a side that is ours and another that is external. But you are right, it is a good idea